src/cli/add.rs

```diff
@@ -380,18 +380,26 @@ fn update_conda_specs_from_lock_file(
         .flatten()
         .collect_vec();
 
-    let pinning_strategy = project.config().pinning_strategy.unwrap_or_default();
+    let mut pinning_strategy = project.config().pinning_strategy;
     let channel_config = project.channel_config();
     for (name, (spec_type, spec)) in conda_specs_to_add_constraints_for {
-        let version_constraint = pinning_strategy.determine_version_constraint(
-            conda_records.iter().filter_map(|record| {
+        // Edge case: python is a special case where we want to pin the minor version by default.
+        // This is done to avoid early user confusion when the minor version changes and environments magically start breaking.
+        // This move a `>=3.13, <4` to a `>=3.13, <3.14` constraint.
```

Comment:
Might be worth adding a docs entry for this

Reply:
Done